libraries/libstratosphere/include/stratosphere/lm/detail/lm_detail_log_packet_transmitter.hpp

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (c) 2018-2020 Atmosphère-NX
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms and conditions of the GNU General Public License,
+ * version 2, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+#include <stratosphere/lm/detail/lm_detail_log_types.hpp>
+
+namespace ams::lm::detail {
+
+    class LogPacketTransmitterBase {
+        NON_COPYABLE(LogPacketTransmitterBase);
+        NON_MOVEABLE(LogPacketTransmitterBase);
+        public:
+            using FlushFunction = bool (*)(const void*, size_t);
+        private:
+            LogPacketHeader *header;
+            u8 *log_buffer_start;
```

Comment:
Kind of verbose, consider e.g. *start; *end; *payload; *current;

This is not a required change.